documentation/tutorial.md

```diff
@@ -68,3 +78,660 @@ tutorial overlord    localhost/localhost  2.9.37   unsupported  23:20:53Z
+
+### MongoDB URI
+Connecting to the database requires a [URI](https://www.mongodb.com/docs/manual/reference/connection-string/). We use a URI of the format:
+```shell
+mongodb://<username>:<password>@<hosts>/<database name>?replicaSet=<replica set name>
+```
+
+Connecting via the URI requires that you know the values for `username`, `password`, `hosts`, `database name`, and the `replica set name`. We will show you how to retrieve the necessary fields. 
+
+**Retrieving the username:** In this case, we are using the `admin` user to connect to MongoDB. Use `admin` as the username.
+
+**Retrieving the password:** The password can be retrieved by running the `get-password` action on the Charmed MongoDB application:
+```shell
+juju run-action mongodb/leader get-password --wait
+```
+Running the command should output:
+```yaml
+unit-mongodb-0:
+  UnitId: mongodb/0
+  id: "2"
+  results:
+    admin-password: <password>
+  status: completed
+  timing:
+    completed: 2022-12-02 11:30:01 +0000 UTC
+    enqueued: 2022-12-02 11:29:57 +0000 UTC
+    started: 2022-12-02 11:30:01 +0000 UTC
+```
+Use the password under the result: `admin-password`.
+
+**Retrieving the hosts:** The hosts are the units hosting the MongoDB application. The host’s IP address can be found with `juju status`:
+
+```
+Model     Controller  Cloud/Region         Version  SLA          Timestamp
+tutorial  overlord    localhost/localhost  2.9.37   unsupported  11:31:16Z
+
+App      Version  Status  Scale  Charm    Channel   Rev  Exposed  Message
+mongodb           active      1  mongodb  dpe/edge   96  no       Replica set primary
+
+Unit        Workload  Agent  Machine  Public address  Ports      Message
+mongodb/0*  active    idle   0        <host IP>    27017/tcp  Replica set primary
+
+Machine  State    Address       Inst id        Series  AZ  Message
+0        started  10.23.62.156  juju-d35d30-0  focal       Running
+
+```
+Use the IP address listed underneath `Public address` for `mongodb/0` as your host.
+
+**Retrieving the database name:** In this case we are connecting to the `admin` database. Use `admin` as the database name. Once we access the database via the MongoDB URI, we will create a `test-db` database to store data.
+
+**Retrieving the replica set name:** The replica set name is the name of the application on Juju hosting MongoDB. The application name in this tutorial is `mongodb`. Use `mongodb` as the replica set name. 
+
+### Connect via MongoDB URI
+Now that we have the necessary fields to connect to the URI, we can connect to MongoDB via the URI. Enter the following into the command line, replace the values for  `username`, `password`, `hosts`, `database name`, and the `replica set name` with what you’ve retrieved above:
```

Comment:
I got blocked at this point, and I think I might have either misunderstood the instructions or just done something really obviously wrong. It might be helpful to the user to have an example to cross-reference again to make sure that they have the format correct?

![image](https://user-images.githubusercontent.com/116022293/208161480-99f68bbf-3f4a-4da8-bd36-0507de191327.png)